src/store/ChartStore.js

```diff
@@ -279,6 +280,18 @@ class ChartStore {
 
         stxx.append('createDataSet', this.updateComparisons);
     }
+    initialRouting() {
+        let timer;
+        window.addEventListener('hashchange', () => {
+            clearTimeout(timer);
+            timer = setTimeout(() => {
+                const hash = window.location.hash.replace('#', '');
```

Comment:
Use `window.history` api, you probably won't need the timeOut

Reply:
timeout is for another purpose. on each dialog opening, I clean the `urlpath` then fill it. because if user press back button, I should recognize that he want to close dialog(by checking the urlpath and compare with empty string).